MaxInterval: trigger "panic: runtime error: integer divide by zero" if set MaxInterval to 0

panic stacktrace
```
panic: runtime error: integer divide by zero

goroutine 581 [running]:
github.com/gounknown/logrotate.evalCurrRotationTime({0x3f3ec60?, 0x853a4d0?}, 0x7080, 0x0)
	/data/home/user00/go/mod/cache/github.com/gounknown/logrotate@v0.0.0-20240409145248-b7726087033e/util.go:73 +0xb2
github.com/gounknown/logrotate.(*Logger).evalCurrentFilename(0xc002fd3b80, 0x97, 0x0)
	/data/home/user00/go/mod/cache/github.com/gounknown/logrotate@v0.0.0-20240409145248-b7726087033e/logrotate.go:305 +0xec
github.com/gounknown/logrotate.(*Logger).openExistingOrNew(0xc002fd3b80, 0x97)
	/data/home/user00/go/mod/cache/github.com/gounknown/logrotate@v0.0.0-20240409145248-b7726087033e/logrotate.go:256 +0x85
github.com/gounknown/logrotate.(*Logger).write(0xc002fd3b80, {0xc009480000, 0x97, 0x80000})
	/data/home/user00/go/mod/cache/github.com/gounknown/logrotate@v0.0.0-20240409145248-b7726087033e/logrotate.go:108 +0x86
github.com/gounknown/logrotate.(*Logger).Write(0xc002fd3b80, {0xc009480000, 0x97, 0x80000})
	/data/home/user00/go/mod/cache/github.com/gounknown/logrotate@v0.0.0-20240409145248-b7726087033e/logrotate.go:99 +0xaf
bufio.(*Writer).Flush(0xc009239c40)
	/usr/local/go/src/bufio/bufio.go:628 +0x62
go.uber.org/zap/zapcore.(*BufferedWriteSyncer).Sync(0xc0020b28a0)
	/data/home/user00/go/mod/cache/go.uber.org/zap@v1.24.0/zapcore/buffered_write_syncer.go:164 +0x9d
go.uber.org/zap/zapcore.(*BufferedWriteSyncer).flushLoop(0xc0020b28a0)
	/data/home/user00/go/mod/cache/go.uber.org/zap@v1.24.0/zapcore/buffered_write_syncer.go:181 +0x6a
created by go.uber.org/zap/zapcore.(*BufferedWriteSyncer).initialize
	/data/home/user00/go/mod/cache/go.uber.org/zap@v1.24.0/zapcore/buffered_write_syncer.go:132 +0x26a
```

--- a/logrotate.go
+++ b/logrotate.go
@@ -354,7 +354,14 @@ func (l *Logger) openNew(filename string) error {
 func (l *Logger) evalCurrentFilename(writeLen int64, forceNewFile bool) string {
 	baseFilename := l.currBaseFilename
 	if l.currBaseFilename == "" {
-		l.currRotationTime = evalCurrRotationTime(l.opts.clock, l.tzOffsetSeconds, l.maxIntervalSeconds)
+		// init base filename if l.currBaseFilename not set
+		if l.maxIntervalSeconds > 0 {
+			l.currRotationTime = evalCurrRotationTime(l.opts.clock, l.tzOffsetSeconds, l.maxIntervalSeconds)
+		} else if l.currRotationTime == 0 {
+			// no rotation based on MaxInterval, just set currRotationTime
+			// to now only once if not set.
+			l.currRotationTime = l.opts.clock.Now().Unix()
+		}
 		baseFilename = genBaseFilename(l.pattern, l.opts.clock, l.currRotationTime)
 	} else if l.maxIntervalSeconds > 0 {
 		rotationTime := evalCurrRotationTime(l.opts.clock, l.tzOffsetSeconds, l.maxIntervalSeconds)

--- a/logrotate_test.go
+++ b/logrotate_test.go
@@ -39,8 +39,8 @@ func Benchmark_MaxBackups1000(b *testing.B) {
 	// time.Sleep(time.Second)
 }
 
-func Benchmark_MaxInterval1Second(b *testing.B) {
-	dir := filepath.Join(baseTestDir, "Benchmark_MaxInterval1Second")
+func Benchmark_MaxInterval(b *testing.B) {
+	dir := filepath.Join(baseTestDir, "Benchmark_MaxInterval")
 	defer os.RemoveAll(dir)
 	l, err := New(filepath.Join(dir, "log%Y%m%d%H%M%S"),
 		WithSymlink(filepath.Join(dir, "log")),
@@ -49,26 +49,6 @@ func Benchmark_MaxInterval1Second(b *testing.B) {
 		// WithMaxAge(3*time.Second),
 		WithMaxBackups(10),
 		WithBufferedWrite(100),
-	)
-	require.NoError(b, err, "New should succeed")
-	defer l.Close()
-
-	log.SetOutput(l)
-
-	logline := "Hello, World"
-	for i := 0; i < b.N; i++ {
-		log.Println(logline)
-	}
-}
-
-func Benchmark_MaxInterval0(b *testing.B) {
-	dir := filepath.Join(baseTestDir, "Benchmark_MaxInterval0")
-	defer os.RemoveAll(dir)
-	l, err := New(filepath.Join(dir, "log%Y%m%d%H%M%S"),
-		WithSymlink(filepath.Join(dir, "log")),
-		WithMaxSize(10),
-		WithMaxInterval(0),
-		WithMaxBackups(10),
 	)
 	require.NoError(b, err, "New should succeed")
 	defer l.Close()
@@ -254,6 +234,20 @@ func Test_Rotate(t *testing.T) {
 				}
 			}
 		})
+	}
+}
+
+func Test_MaxInterval0(t *testing.T) {
+	dir := filepath.Join(baseTestDir, "Benchmark_MaxInterval0")
+	defer os.RemoveAll(dir)
+	l, err := New(filepath.Join(dir, "log%Y%m%d%H%M%S"),
+		WithSymlink(filepath.Join(dir, "log")),
+		WithMaxSize(10),
+		WithMaxInterval(0),
+	)
+	require.NoError(t, err, `New should succeed`)
+	for i := 0; i < 10; i++ {
+		l.Write([]byte("Hello, World"))
 	}
 }
 

--- a/logrotate_test.go
+++ b/logrotate_test.go
@@ -39,8 +39,8 @@ func Benchmark_MaxBackups1000(b *testing.B) {
 	// time.Sleep(time.Second)
 }
 
-func Benchmark_MaxInterval(b *testing.B) {
-	dir := filepath.Join(baseTestDir, "Benchmark_MaxInterval")
+func Benchmark_MaxInterval1Second(b *testing.B) {
+	dir := filepath.Join(baseTestDir, "Benchmark_MaxInterval1Second")
 	defer os.RemoveAll(dir)
 	l, err := New(filepath.Join(dir, "log%Y%m%d%H%M%S"),
 		WithSymlink(filepath.Join(dir, "log")),
@@ -49,6 +49,26 @@ func Benchmark_MaxInterval(b *testing.B) {
 		// WithMaxAge(3*time.Second),
 		WithMaxBackups(10),
 		WithBufferedWrite(100),
+	)
+	require.NoError(b, err, "New should succeed")
+	defer l.Close()
+
+	log.SetOutput(l)
+
+	logline := "Hello, World"
+	for i := 0; i < b.N; i++ {
+		log.Println(logline)
+	}
+}
+
+func Benchmark_MaxInterval0(b *testing.B) {
+	dir := filepath.Join(baseTestDir, "Benchmark_MaxInterval0")
+	defer os.RemoveAll(dir)
+	l, err := New(filepath.Join(dir, "log%Y%m%d%H%M%S"),
+		WithSymlink(filepath.Join(dir, "log")),
+		WithMaxSize(10),
+		WithMaxInterval(0),
+		WithMaxBackups(10),
 	)
 	require.NoError(b, err, "New should succeed")
 	defer l.Close()

